photon-client/src/integTest/scala/com/linkedin/photon/ml/estimators/GameEstimatorTest.scala

```diff
@@ -65,14 +249,20 @@ class GameEstimatorTest extends SparkTestUtils with GameTestUtils {
   @Test
   def testPrepareFixedAndRandomEffectTrainingDataSet(): Unit =
     sparkTest("prepareFixedAndRandomEffectTrainingDataSet", useKryo = true) {
+
       val featureSectionMap = fixedAndRandomEffectFeatureSectionMap
       val data = getData(trainPath, featureSectionMap)
-      val trainingDataSet = getEstimator(fixedAndRandomEffectParams).prepareTrainingDataSet(
-        GameConverters.getGameDataSetFromDataFrame(
-          data,
-          featureSectionMap.keys.toSet,
-          idTypeSet,
-          isResponseRequired = true).partitionBy(new LongHashPartitioner(data.rdd.partitions.length)))
+      val partitioner = new LongHashPartitioner(data.rdd.partitions.length)
+      val gameDataSet = GameConverters.getGameDataSetFromDataFrame(
+        data,
+        featureSectionMap.keys.toSet,
+        idTypeSet,
+        isResponseRequired = true)
+        .partitionBy(partitioner)
+
+      val params = fixedAndRandomEffectParams
+      val (estimator, _) = createEstimator(params, "prepareFixedAndRandomEffectTrainingDataSet")
+      val trainingDataSet = estimator.prepareTrainingDataSet(gameDataSet)
```

Comment:
Any reason for this block of changes? Usually you like to compress code blocks to be more functional, not less